guides/plugins/plugins/storefront/add-scss-variables.md

```diff
@@ -0,0 +1,212 @@
+
+To add SCSS variables to your plugin you can use a subscriber class to add custom SCSS variables.
+
+## Prerequisites
+
+You won't learn how to create a plugin in this guide, head over to our Plugin base guide to create
+your first plugin.
+
+{% page-ref page="./../plugin-base-guide.md" %}
+
+## Setup a default value for a custom SCSS variable
+
+Before you start adding your subscriber you should provide a fallback value for your custom SCSS variable in your plugin `base.scss`:
+
+{% code title="<plugin root>/src/Resources/app/storefront/src/scss/base.scss" %}
+```scss
+// The value will be overwritten by the subscriber when the plugin is installed and activated
+$sass-plugin-header-bg-color: #ffcc00 !default;
+
+.header-main {
+    background-color: $sass-plugin-header-bg-color;
+}
+```
+{% endcode %}
+
+## Theme variables subscriber
+
+You can add a new subscriber according to the [Listening to events](../plugin-fundamentals/listening-to-events.md) guide.
+In this example we name the subscriber `ThemeVariableSubscriber.php`. The subscriber listens to the `ThemeCompilerEnrichScssVariablesEvent`.
+
+{% tabs %}
+{% tab title="<plugin root>/src/Subscriber/ThemeVariablesSubscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Shopware\Storefront\Event\ThemeCompilerEnrichScssVariablesEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ThemeVariablesSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ThemeCompilerEnrichScssVariablesEvent::class => 'onAddVariables'
+        ];
+    }
+
+    public function onAddVariables(ThemeCompilerEnrichScssVariablesEvent $event)
+    {
+        // Will render: $sass-plugin-header-bg-color: "#59ccff";
+        $event->addVariable('sass-plugin-header-bg-color', '#59ccff');
+    }
+}
+```
+{% endtab %}
+{% tab title="<plugin root>/src/Resources/config/services.xml" %}
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    
+    <services>
+        <service id="Swag\BasicExample\Subscriber\ThemeVariablesSubscriber">
+            <tag name="kernel.event_subscriber"/>
+        </service>
+    </services>
+</container>
+```
+{% endtab %}
+{% endtabs %}
+
+The `ThemeCompilerEnrichScssVariablesEvent` provides the `addVariable()` method which takes the following parameters:
+
+* `$name:` (string): The name of the SCSS variable. The passed string will be exactly in your SCSS so please be careful with special characters. We recommend using kebab-case here. The variable prefix `$` will be added automatically. We also recommend prefixing your variable name with your plugin's or company's name to prevent naming conflicts.
```

Comment:
What do you think about using a table here?

Reply:
I do think this style suffices, as long as the list doesn't get too long. But sure, he can change it if he wants to

Reply:
A table would remove the option for the following lines to break below the key, that gets described, and therefore rendering the description even less readable. Vote against table

Reply:
Alright, it is set then 👍 We'll leave it as it is.